components/PageComponents.vue

```diff
@@ -27,10 +31,11 @@ export default {
       data: { data },
     } = await axios.get(`${process.env.strapiUrl}/api/${this.endpoint}`, {
       params: {
+        ...this.params,
         populate: 'deep',
       },
     })
-    this.components = data.components
+    this.components = data.components || data[0].components
```

Comment:
Why data[0] in fallback

Reply:
So When we add filter in params it returns an array. But if we don't add filter it gives an object for single pages like Home Page. So when it gives an array instead of object then the fallback will be used.